File: Programming Skills II/66. Plus One/66. Plus One.java
class Solution {
    public int[] plusOne(int[] digits) {
        int n = digits.length;
        for(int i=n-1; i>=0; --i){
            if( digits[i] != 9 ){
                digits[i]++;
                return digits;
            }else{
                digits[i]=0;
            }
        }
        int [] ans = new int[n+1];
        ans [0] = 1;
        int i = 1;
        for(int elm : digits ){
            ans [i] = elm;
            i++;
        }
        return ans;
    }
}

File: Programming Skills II/66. Plus One/66. Plus One.cpp
class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
        int n = digits.size();
        int i = n-1;
        for(; i>=0; --i){
            if(digits[i] != 9){
                digits[i]++;
                return digits;
            }else{
                digits[i] = 0;
            }
        }
        vector<int> ans;
        ans.push_back(1);
        ans.insert(ans.end(), digits.begin(), digits.end());
        return ans;
    }
};
